File: DeckOfCards-Scala/src/main/scala/com/arctouch/deckofcards/Card.scala
package com.arctouch.deckofcards

object Suit extends Enumeration {
  type Suit = Value
  val Clubs, Diamonds, Hearts, Spades = Value
}

object Rank extends Enumeration {
  type Rank = Value
  val Ace, Two, Three, Four, Five, Six, Seven, Eight, Nine, Ten, Jack, Queen, King = Value
}

import Suit._
import Rank._

case class Card(suit: Suit, rank: Rank) {
  override def toString() = {
    s"${rank} of ${suit}"
  }
}

File: DeckOfCards-Ruby/src/card.rb
SUITS = %i[Hearts Clubs Diamonds Spades]
RANKS = %i[Ace Two Three Four Five Six Seven Eight Nine Ten Jack Queen King]

class Card
  attr_reader :suit, :rank

  def initialize(suit, rank)
    @rank = rank
    @suit = suit
  end

  def ==(another_card)
    suit == another_card.suit && rank == another_card.rank
  end

  def to_s
    "#{@rank} of #{@suit}"
  end

  def inspect
    "#<#{@rank} of #{@suit}>"
  end
end
